### PR TITLE
[Fixes #9604] Report empty thumbnail in case a thumbnail has not been generated for a resource

### LINKED ISSUE #9604: Report empty thumbnail in case a thumbnail has not been generated for a resource
At the moment the API returns the link to the default empty thumbnail image, in case a real thumbnail wasn't generated for a resource.

We want instead a null value to be returned in that case, and let the client know that no thumbnail image is available for the resource. It will take care of showing a placeholder or a message.

This logic will be implemented for any resource type.

### PR DESCRIPTION
**Pull request by marthamareal**
<Include a few sentences describing the overall goals for this Pull Request>

References #9604 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by afabiani** [comment]
@marthamareal this PR keeps failing on CircleCI, but I cannot exactly understand why... it looks like the containers do not start correctly. Can you please double check?

**Comment by marthamareal** [comment]
> @marthamareal this PR keeps failing on CircleCI, but I cannot exactly understand why... it looks like the containers do not start correctly. Can you please double check?

Okay

File: geonode/base/migrations/0082_clean_resources_with_missing_thumb.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    try:
+        resource_model = apps.get_model('base', 'ResourceBase')
+        link_model = apps.get_model('base', 'Link')
+        # update thumbnail urls
+        resource_model.objects.filter(thumbnail_url__icontains=static(MISSING_THUMB)).update(thumbnail_url=None)
+        # Remove thumbnail links
+        link_model.objects.filter(resource__thumbnail_url__isnull=True, name='Thumbnail').delete()
+    except Exception as e:
+        logger.error(f'Error during cleanup of missing thumbnails and links: {e.args[0]}')
+        pass
```
Comment (afabiani) [inline geonode/base/migrations/0082_clean_resources_with_missing_thumb.py:22]: Please, remove the `pass` statement here and switch `logger.error(f'Error during cleanup of missing thumbnails and links: {e.args[0]}')` to `logger.exception(e)`

**Review by afabiani — CHANGES_REQUESTED** [review]

**Comment by afabiani** [comment]
@marthamareal the build is failing

**Comment by afabiani** [comment]
@marthamareal CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0082_remove_dialogos_comment, 0083_clean_resources_with_missing_thumb in base).

**Review by afabiani — APPROVED** [review]

**Pull request merged**